src/sql-parser/tests/sqlparser_common.rs

```diff
@@ -104,6 +105,37 @@ Expected INTO, found: public"
     );
 }
 
+#[test]
+fn op_precedence() {
+    struct RemoveParens;
+
+    impl<'a> VisitMut<'a> for RemoveParens {
+        fn visit_expr(&mut self, expr: &'a mut Expr) {
+            if let Expr::Nested(e) = expr {
+                *expr = (**e).clone();
+            }
+            visit_mut::visit_expr(self, expr);
+        }
+    }
+
+    for (actual, expected) in &[
+        ("a + b + c", "(a + b) + c"),
+        ("a - b + c", "(a - b) + c"),
+        ("a + b * c", "a + (b * c)"),
+        ("true = 'foo' like 'foo'", "true = ('foo' like 'foo')"),
+        ("a->b = c->d", "(a->b) = (c->d)"),
+        ("a @> b = c @> d", "(a @> b) = (c @> d)"),
+        ("a = b is null", "(a = b) is null"),
+        ("a and b or c and d", "(a and b) or (c and d)"),
```

Comment:
Ha! This is clever. There's a bunch of uglier precedence tests scattered throughout this file that perhaps you'd like to absorb here?